jina/flow/base.py

```diff
@@ -981,10 +981,8 @@ def rolling_update_thread(self, pod_name):
 
         :param pod_name: pod to update
         """
-        print('start thread')
         x = threading.Thread(target=self.rolling_update, args=(pod_name,))
```

Comment:
I think it should not be in a Thread. It should be blocking. After all, the Runtime receiving queries is in another Process, so it makes no sense inmho to add a thread here

Reply:
If we don't use a thread, we can not run queries while reloading the replicas.
Another way of user interaction would be to start a flow server:
```
with flow:
    flow.block()
```
And then send data requests to the flow as well as rolling update requests.
This can be implemented in the close future.

I could move the threading to the in the user code in the test for now.

Reply:
the problem is the need of a Flow object to wrap a client. The proper way should be to instantiate a client outside, and yes moving the thread in the user code should be the way to go

Reply:
okay done